### ISSUE TITLE
Build provider functionbeat executables for linux/amd64

### ISSUE THREAD

**Issue by ycombinator** [elastic/beats#15132]
Currently, when `mage build` is run from the `x-pack/functionbeat` folder, it builds two executables: `x-pack/functionbeat/functionbeat` (the manager, which is intended to be executed locally) and `x-pack/functionbeat/providers/aws/functionbeat-aws` (the executable to be invoked by the serverless provider, AWS in this case).

The former should be built for the local platform but the latter should be built for `linux/amd64` as that will be the platform used in the serverless environment, at least in AWS (and that's the only provider we have at the moment).

So we should adjust the `build` target in the `magefile.go` accordingly.

**Comment by ycombinator** [comment]
Workaround:

```
cd $GOROOT/src/github.com/elastic/beats/x-pack/functionbeat/providers/aws
GOOS=linux go build -o functionbeat-aws main.go
```

**Comment by rvillablanca** [comment]
Friendly reminder this is done

**Comment by ycombinator** [comment]
Closed via #15586.
